ui/hooks/snaps/useDisplayName.ts

```diff
@@ -0,0 +1,52 @@
+import { NamespaceId } from '@metamask/snaps-utils';
+import { CaipChainId, KnownCaipNamespace } from '@metamask/utils';
+import { useSelector } from 'react-redux';
+import {
+  getMemoizedAccountName,
+  getAddressBookEntryByNetwork,
+} from '../../selectors';
+import { toChecksumHexAddress } from '../../../shared/modules/hexstring-utils';
+import { decimalToHex } from '../../../shared/modules/conversion.utils';
+
+export type UseDisplayNameParams = {
+  chain: {
+    namespace: NamespaceId;
+    reference: string;
+  };
+  chainId: CaipChainId;
+  address: string;
+};
+
+/**
+ * Get the display name for an address.
+ * This will look for an account name in the state, and if not found, it will look for an address book entry.
+ *
+ * @param params - The parsed CAIP-10 ID.
+ * @returns The display name for the address.
+ */
+export const useDisplayName = (params: UseDisplayNameParams): string => {
+  const {
+    address,
+    chain: { namespace, reference },
+  } = params;
+
+  const isEip155 = namespace === KnownCaipNamespace.Eip155;
+
+  const parsedAddress = isEip155 ? toChecksumHexAddress(address) : address;
+
+  const accountName = useSelector((state) =>
+    // @ts-expect-error type issue
```

Comment:
Since we are adding new selectors can we add them with types so we don't have to do this?